cpp/src/arrow/array/array_dict_test.cc

```diff
@@ -1499,6 +1503,35 @@ TEST(TestDictionary, Compact) {
   }
 }
 
+TEST(TestDictionary, CompactWithCompleteDictionary) {
```

Comment:
I'm not sure what this test adds compared to the existing ones above. Can you explain or comment on it?